documentation/docs/requirements/use-case-descriptions.md

```diff
@@ -5,65 +5,89 @@ sidebar_position: 5
+III) Player wants to challenge themselves and selects hard options while playing:
+
+    1) Player selects option Story Mode/Single Player.
+    2) Player selects an unlocked level.
+    3) A level is generated for the player.
+    4) Player selects the timer option.  
+    5) Player has a certain time to complete the level.
+
+IV) Player wants to play a game with a friend:
+
+    1)  Player selects the Multiplayer option. 
+    2)  Player selects the Co-op option. 
+    3)  Player sets up a lobby by creating a room and tells their friend the name of the room to join.
+    4)  Both players are on the same map.
+    5)  Players are helping each other in order to win the level.
+
+V) Player wants to play a game against a friend:
+
+    1)  Player selects the Multiplayer option. 
+    2)  Player selects the Compete option. 
+    3)  Player sets up a lobby by creating a room and tells their friend the name of the room to join.
+    4)  Players are competing on the same level in a split-screen.
+    5)	Player gets the boxes into the end point faster than their friend.
+    6)  Player won the game.
+
+VI) Player wants to check high scores:
+
+    1)	Player selects the High Scores option.
+    2)	Player views high scores.
+
+VII) Game saves player's progress:
+
+    1)	Player selects Story Mode/Single Player option.
+    2)	Player passes the first level, and their progress is saved automatically.
+    3)	Player exits the game by pressing the “X” button.
+    4)	Player opens the game again and logs into their account.
+    5)	Player selects the Story Mode/Single Player option.
+    6)	Player can start now from Level 2 because their progress was saved.
+
+VIII) Player wins level:
+
+    1)	Player selects Story Mode/Single Player option.
+    2)	Player selects an unlocked level.
+    3)  A level is generated for the player.
+    4)  Player starts to move the character and push the box by using the arrows on the keyboard.
+    5)	Player places the box onto the target.
+    6)	Player wins the level.
+
+IX) Player gets stuck and restarts the level:
+
+    1)	Player selects Story Mode/Single Player option.
+    2)	Player moves the character and pushes the box(es) trying to win the level.
+    3)	Player realizes that got stuck by pushing a box onto the wall and won’t be able to place the box onto the target.
+    4)	Player restarts the level by pressing the spacebar.
+    5)	The level restarts with the character and the box(es) back to their starting positions.
+
+X) Player wants to play with a different character:
+
+    1)  Player selects Story Mode/Single Player option.
+    2)  Player selects their favorite character from a variety of characters.
+    3)  Player starts playing with the chosen character.
+
+XI) Player wants to challenge themselves onto a map with complex design.
+
+    1)  Player selects Story Mode/Single Player option.
+    2)  Player selects from a variety of maps with complex deisgn.
```

Comment:
Shouldn't the player be given a random procedural generated difficult map rather than a selection of maps?

Reply:
Wait actually whats the difference between this and Use Case 3?

Reply:
This is actually a non functional requirement. I just had to make a use case for it.